Rollbar/RollbarTelemetry.h

```diff
@@ -0,0 +1,46 @@
+//
+//  RollbarTelemetry.h
+//  Rollbar
+//
+//  Created by Ben Wong on 11/21/17.
+//  Copyright © 2017 Rollbar. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "RollbarLevel.h"
+#import "RollbarTelemetryType.h"
+
+#define NSLog(args...) [RollbarTelemetry NSLogReplacement:args];
+
+@interface RollbarTelemetry : NSObject {
+    NSMutableArray *dataArray;
+    NSInteger limit;
+    NSString *dataFilePath;
+}
+
++ (instancetype)sharedInstance;
++ (void)NSLogReplacement:(NSString *)format, ...;
+
+#pragma mark -
+
+- (void)setCaptureLog:(BOOL)shouldCapture;
+- (void)setDataLimit:(NSInteger)dataLimit;
+
+#pragma mark -
+
+- (void)recordEventForLevel:(RollbarLevel)level type:(RollbarTelemetryType)type data:(NSDictionary *)data;
+
+- (void)recordDomEventForLevel:(RollbarLevel)level element:(NSString *)element extraData:(NSDictionary *)extraData;
```

Comment:
I wish we hadn't called this dom in the spec, maybe we can call it something like `recordViewEvent...` here and only refer to it as dom under the hood when we need to make it an object that will get sent to the api

Reply:
Renamed to recordViewEvent